addons/stock/models/stock_picking.py

```diff
@@ -312,13 +312,13 @@ class Picking(models.Model):
     location_id = fields.Many2one(
         'stock.location', "Source Location",
         default=lambda self: self.env['stock.picking.type'].browse(self._context.get('default_picking_type_id')).default_location_src_id,
-        check_company=True, readonly=True, required=True,
-        states={'draft': [('readonly', False)]})
+        check_company=True, readonly=False, required=True,
```

Comment:
You can remove the `readonly=False` here and in `location_dest_id` since it is the default